spouse/spouse_demo.py

```diff
@@ -0,0 +1,363 @@
+# %% [markdown]
+# # Detecting spouse mentions in sentences
+
+# %% [markdown]
+# We will walk through an example text classification task to explore how Snorkel can be used for Information Extraction.
+# ## Classification Task
+# <img src="imgs/sentence.jpg" width="700px;">
+#
+# We want to classify each __candidate__ or pair of people mentioned in a sentence, as being married at some point or not.
+#
+# In the above example, our candidate represents the possible relation `(Barack Obama, Michelle Obama)`. As readers, we know this mention is true due to external knowledge and the keyword of `wedding` occuring later in the sentence.
+# We begin with some basic setup and data downloading.
+#
+# %%
+# %matplotlib inline
+
+import os
+import pickle
+import numpy as np
+
+if os.path.basename(os.getcwd()) == "snorkel-tutorials":
+    os.chdir("spouse")
+
+from utils import load_data
+
+((df_dev, Y_dev), df_train, (df_test, Y_test)) = load_data()
+
+# %% [markdown]
+# **Input Data:** `df_dev` is a Pandas DataFrame object, where each row represents a particular __candidate__. The DataFrames contain the fields `sentence`, which refers to the sentence the candidate is in, `tokens`, the tokenized form of the sentence, `person1_word_idx` and `person2_word_idx`, which represent `[start, end]` indices in the tokens at which the first and second person's name appear, respectively.
+#
+# We also have certain **preprocessed fields**, that we discuss a few cells below.
+
+# %%
+import pandas as pd
+
+# Don't truncate text fields in the display
+pd.set_option("display.max_colwidth", 0)
+
+df_dev.head()
+
+# %% [markdown]
+# Let's look at a candidate in the development set:
+
+# %%
+from preprocessors import get_person_text
+
+candidate = df_dev.loc[2]
+person_names = get_person_text(candidate).person_names
+
+print("Sentence: ", candidate["sentence"])
+print("Person 1: ", person_names[0])
+print("Person 2: ", person_names[1])
+
+# %% [markdown]
+# ### Preprocessing the Database
+#
+# In a real application, there is a lot of data preparation, parsing, and database loading that needs to be completed before we generate candidates and dive into writing labeling functions. Here we've pre-generated candidates in a pandas DataFrame object per split (train,dev,test).
```

Comment:
Let's not overload "operators", because that's what we've been calling LFs/TFs/SFs.
Maybe we can allude to these as intuitive patterns, or something to that effect.